### ISSUE TITLE
Remove ReadSeeker assertions

### ISSUE THREAD

**Issue by awused** [faiface/beep#25]
I want to use this to play a stream over HTTP, but the library only supports ReadClosers that are also ReadSeekers. If I comment out the type assertion [here](https://github.com/faiface/beep/blob/master/flac/decode.go#L24) I'm able to play a flac stream from my other machine without any problems.

Calling any Seek() methods without seek support should return an error (not cause a panic), but the library shouldn't require it when users don't call those methods.

**Comment by awused** [comment]
I'm not sure what you mean by redundant. A ReadCloser does not contain all methods that a ReadSeeker contains, so the assertion cannot be redundant since there are ReadClosers that are not ReadSeekers. Runtime checks and panics across package boundaries are also a bad practice.

In the future instead of type assertions and panics, you can define your own interface so you'll get compile time checks.

```go
type ReadSeekCloser interface {
  io.ReadSeeker
  io.Closer
}
```

I ended up abandoning this little project and rewriting it in C++ for unrelated reasons. One of the libraries, failed to play back FLAC without glitches even with local files with the newest Go version. I didn't feel like debugging it, so I can't even say which dependency it was.

**Comment by mewmew** [comment]
> One of the libraries, failed to play back FLAC without glitches even with local files with the newest Go version.

@awused Would you mind sharing the FLAC file you used?

Cheers,
/u

**Comment by awused** [comment]
I didn't find that it was caused by any specific file, and wasn't consistently reproducible at any one point in the stream or files. They tended to happen every few minutes, but I never found any patterns in the input that could reliably trigger them. I never tried anything that wasn't flac, so I can't even tell you if it's specific to flac.

The majority of my library is 16bit 44.1khz encoded with level 8 compression by the reference flac encoder. The stream specifically was produced by MPD using the latest reference flac with level 0 compression (this is my local network) at 24bit/48khz. Buffer size did not help, even with excessively large buffers, though I didn't play as much with local files as I did the stream. Playback was done on Windows 10 64bit with audio output set to 24bit/48khz, minimal system load, and a local NVMe SSD.
